src/Stratis.Bitcoin.Features.RPC/RpcSettings.cs

```diff
@@ -32,74 +32,118 @@ public class RpcSettings
         {
             Guard.NotNull(nodeSettings, nameof(nodeSettings));
 
             this.logger = nodeSettings.LoggerFactory.CreateLogger(typeof(RpcSettings).FullName);
 
+            this.RPCPort = nodeSettings.Network.DefaultRPCPort;
             this.Bind = new List<IPEndPoint>();
-            this.DefaultBindings = new List<IPEndPoint>();
             this.AllowIp = new List<IPAddressBlock>();
+            this.Server = false;
+            this.RpcUser = null;
+            this.RpcPassword = null;
 
-            // Get values from config
+            defaultArgs?.Invoke(this);
+
+            // For printing help.
+            // TODO: Refactor.
+            settingsHelp = (RpcSettings)this.MemberwiseClone();
+
+            // Get values from config and command line.
             this.LoadSettingsFromConfig(nodeSettings);
 
             // Check validity of settings
             this.CheckConfigurationValidity(nodeSettings.Logger);
         }
 
         /// <summary>
-        /// Loads the rpc settings from the application configuration.
+        /// Loads the RPC settings from the command line and application configuration.
         /// </summary>
         /// <param name="nodeSettings">Application configuration.</param>
         private void LoadSettingsFromConfig(NodeSettings nodeSettings)
         {
             TextFileConfiguration config = nodeSettings.ConfigReader;
 
-            this.Server = config.GetOrDefault<bool>("server", false, this.logger);
-            this.RPCPort = config.GetOrDefault<int>("rpcport", nodeSettings.Network.DefaultRPCPort, this.logger);
+            this.Server = config.GetOrDefault<bool>("server", this.Server, this.logger);
+            this.RPCPort = config.GetOrDefault<int>("rpcport", this.RPCPort, this.logger);
+            this.RpcUser = config.GetOrDefault<string>("rpcuser", this.RpcUser, this.logger);
+            this.RpcPassword = config.GetOrDefault<string>("rpcpassword", this.RpcPassword); // No logging!
 
-            if (this.Server)
+            try
             {
-                this.RpcUser = config.GetOrDefault<string>("rpcuser", null, this.logger);
-                this.RpcPassword = config.GetOrDefault<string>("rpcpassword", null); // No logging!
+                List<IPAddressBlock> allowIp = config
+                    .GetAll("rpcallowip", this.logger)
+                    .Select(p => IPAddressBlock.Parse(p))
+                    .ToList();
 
-                try
-                {
-                    this.AllowIp = config
-                        .GetAll("rpcallowip", this.logger)
-                        .Select(p => IPAddressBlock.Parse(p))
-                        .ToList();
-                }
-                catch (FormatException)
-                {
-                    throw new ConfigurationException("Invalid rpcallowip value");
-                }
+                if (allowIp.Count != 0)
+                    this.AllowIp = allowIp;
```

Comment:
Why dont you just the collection directly to the above code?

```
this.AllowIp = config.GetOrDefault<string>("rpcpassword", null); // No logging!
                    .GetAll("rpcallowip", this.logger)
                    .Select(p => IPAddressBlock.Parse(p))
                    .ToList();
```

Reply:
If you look at the flow in the rpc settings constructor you will see that we (1) assign some constants and then (2) allow those constants to be overridden by the application. If we simply assign here then the application-specific defaults will be overwritten.